lib/rust/parser/tests/parse.rs

```diff
@@ -361,6 +361,55 @@ fn plus_negative() {
     test(&code.join("\n"), expected);
 }
 
+#[test]
+fn minus_binary() {
+    let cases = [
+        ("x - 5", block![(OprApp (Ident x) (Ok "-") (Number 5))]),
+        ("x-5", block![(OprApp (Ident x) (Ok "-") (Number 5))]),
+        ("x.-y", block![(OprApp (Ident x) (Ok ".") (UnaryOprApp "-" (Ident y)))]),
+        ("x.~y", block![(OprApp (Ident x) (Ok ".") (UnaryOprApp "~" (Ident y)))]),
+    ];
+    cases.into_iter().for_each(|(code, expected)| test(code, expected));
+}
+
+#[test]
+fn minus_section() {
+    #[rustfmt::skip]
+    let cases = [
+        ("- x", block![(OprSectionBoundary (OprApp () (Ok "-") (Ident x)))]),
+        ("(- x)", block![
+            (MultiSegmentApp #(
+             ((Symbol "(") (OprSectionBoundary (OprApp () (Ok "-") (Ident x))))
+             ((Symbol ")") ())))]),
+        ("- (x * 2)", block![
+            (OprSectionBoundary (OprApp () (Ok "-")
+             (MultiSegmentApp #(
+              ((Symbol "(") (OprApp (Ident x) (Ok "*") (Number 2)))
+              ((Symbol ")") ())))))]),
+    ];
+    cases.into_iter().for_each(|(code, expected)| test(code, expected));
+}
+
+#[test]
+fn minus_unary() {
+    #[rustfmt::skip]
+    let cases = [
+        ("f -5", block![(App (Ident f) (UnaryOprApp "-" (Number 5)))]),
+        ("-5", block![(UnaryOprApp "-" (Number 5))]),
+        ("(-5)", block![
+            (MultiSegmentApp #(((Symbol "(") (UnaryOprApp "-" (Number 5))) ((Symbol ")") ())))]),
+        ("-(x * 2)", block![
+            (UnaryOprApp "-" (MultiSegmentApp #(
+             ((Symbol "(") (OprApp (Ident x) (Ok "*") (Number 2)))
```

Comment:
Out of curiosity. There is `Symbol "("` in the tree. Shouldn't parenthesis disappear and be replaced by the structure of the tree?

Reply:
We need every token represented somewhere, but `MultiSegmentApp` is an obtuse way to do that for parentheses; I am planning to introduce a node type for parenthesized expressions; I might be able to fit that in today.